### PR TITLE
#117 refactoring for notifications

### PR DESCRIPTION
**Pull request by kamuffe**
Added a PR suggestion for issue 117.

- moved all notification classes/interfaces to an own package called notification
- impement against interfaces
- added javadoc

### CODE REVIEW

File: app-layout-addon/src/main/java/com/github/appreciated/app/layout/notification/DefaultNotification.java [hunk 1]
```diff
@@ -12,14 +12,13 @@
...
-
 public class DefaultNotification implements Notification {
-    String title;
-    String description;
-    Resource image;
+    private String title;
+    private String description;
+    private Resource image;
     private Priority priority;
-    private boolean dismissible;
```
Comment (kamuffe) [inline app-layout-addon/src/main/java/com/github/appreciated/app/layout/notification/DefaultNotification.java]: Is "isIgnoreable" a better wording for that?

Comment (appreciated) [reply app-layout-addon/src/main/java/com/github/appreciated/app/layout/notification/DefaultNotification.java]: You are a champ 👍.
> Is "isIgnoreable" a better wording for that?
I think "sticky" is in that case a more common word.

**Comment by kamuffe** [comment]
I would also add the possibility to set the time for notifications. Since we have an own notification mechanism (persistable notifications) and these are created hours or days befor the real notification is displayed in application.

**Comment by kamuffe** [comment]
OK, so it will not hurt if notifications are moved out in a new repo.

**Pull request merged**